From 9ea9e230849a40617f424c94d80d1c0fb2ac2c56 Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Fri, 2 Mar 2018 21:25:36 +0000
Subject: [PATCH 2/3] aloop: add asound loopback driver

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
---
 arch/arm64/configs/defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/configs/defconfig b/arch/arm64/configs/defconfig
index 49584e2..8d409a7 100644
--- a/arch/arm64/configs/defconfig
+++ b/arch/arm64/configs/defconfig
@@ -642,4 +642,5 @@ CONFIG_LEDS_PCA995X=y
 CONFIG_SND_SOC_IMX_TFA98XX=y
 CONFIG_SND_SOC_IMX_SPH0645=y
 CONFIG_SND_SOC_SPH0645=y
+CONFIG_SND_ALOOP=y
 
-- 
2.7.4

