From e12aa0d94fe4777fc622a0a8b1b4cfedef3246d2 Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Fri, 3 Aug 2018 20:30:42 +0000
Subject: [PATCH] Update the AWB header and model names

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
---
 SampleApp/src/PortAudioMicrophoneWrapper.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/SampleApp/src/PortAudioMicrophoneWrapper.cpp b/SampleApp/src/PortAudioMicrophoneWrapper.cpp
index 6d12aaea..a37a28a6 100644
--- a/SampleApp/src/PortAudioMicrophoneWrapper.cpp
+++ b/SampleApp/src/PortAudioMicrophoneWrapper.cpp
@@ -61,14 +61,14 @@ static const unsigned int NUM_MEDIA_CHANNELS = 2;
 static const unsigned int NUM_AVS_CHANNELS = 1;
 
 
-#include "VoiceUI_imx8_2mic_AVS_v3b_1mb_sensory_AWELibSimple.h"
+#include "VoiceUI_imx8_2mic_AmazonWW_AWELibSimple.h"
 static const unsigned int NUM_HARDWARE_INPUT_CHANNELS = 2;
 static const unsigned int NUM_HARDWARE_OUTPUT_CHANNELS = 2;
 static const unsigned int NUM_AWE_INPUT_CHANNELS = 4; // expect an AWE design with 4 input channels
 static const unsigned int NUM_AWE_OUTPUT_CHANNELS = 3; // expect an AWE design with 3 output channels
 static const unsigned int SAMPLE_RATE = 48000; // hardware sample rate
 static const unsigned int BLOCK_SIZE = 768;
-const char awbFilename[] = "/home/root/AWELib/VoiceUI_imx8_2mic_AVS_v3b_1mb_sensory_AWELib.awb";
+const char awbFilename[] = "/home/root/AWELib/VoiceUI_imx8_2mic_AmazonWW_AWELib.awb";
 
 static const unsigned int AVS_SAMPLE_RATE = 16000;
 static const unsigned int DECIMATION_FACTOR = SAMPLE_RATE/AVS_SAMPLE_RATE; // ratio of hardware sample rate to AVS sample rate
-- 
2.13.3

